pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -1070,6 +1071,21 @@ func (os *OpenStack) CreateVolume(name string, size int, tags *map[string]string
 	return vol.ID, err
 }
 
+func (os *OpenStack) GetDevicePath(diskId string) string {
+	files, _ := ioutil.ReadDir("/dev/disk/by-id/")
```

Comment:
For generic openstack: I think it's fine to stick with the same existing code in this change, but I share @metral's concerns about finding unformatted volumes by-uuid.  I like the `CinderProvider.GetDevicePath` structure you've added to this change, and we should use that in a later change to replace the generic OpenStack logic with something that also uses the cinder API to find the device name (looks like this is `GetVolume(diskId).attachments[i].device`, where i==our_server_id in cinder v2 API).

It's probably feasible to try cinder API v2 or v1 depending on what appears to be available, and therefore combine the rackspace+openstack code again.  I'm not fussed about whether we do that in this change or not (there are so many other places where the rackspace+openstack providers could share code, that it seems like that's a reasonable project in itself).